### PR TITLE
[ADH-4406] Implement new action API endpoints

### PR DESCRIPTION
**Pull request by tigrulya-exe**
- Implemented new Action API endpoints
- Added search support to the `ActionDao` using Metastore DSL queries
- Correctly handled cases when the action is not found
- Removed unused code related to actions from `Metastore` and `ActionDao`

### CODE REVIEW

**Comment by tigrulya-exe** [comment]
@arenadata/java

File: smart-engine/src/main/java/org/smartdata/server/engine/cmdlet/CmdletInfoHandler.java [hunk 1]
```diff
@@ -107,6 +109,17 @@ public CmdletInfo getCmdletInfo(long cid) throws IOException {
     }
   }
 
+  public ActionInfo getSingleActionInfo(long cmdletId) throws IOException {
+    Long actionId = Optional.ofNullable(getCmdletInfo(cmdletId))
+        .map(CmdletInfo::getAids)
+        .filter(CollectionUtils::isNotEmpty)
+        .map(actionIds -> actionIds.get(0))
```
Comment (VitekArkhipov) [inline smart-engine/src/main/java/org/smartdata/server/engine/cmdlet/CmdletInfoHandler.java]: same, what about using findFirst()?

**Review by VitekArkhipov — APPROVED** [review]

File: smart-common/src/main/java/org/smartdata/model/ActionInfo.java [hunk 2]
```diff
@@ -15,139 +15,101 @@
...
-  public void setArgsFromJsonString(String jsonArgs) {
-    Gson gson = new Gson();
-    args = gson.fromJson(jsonArgs,
-        new TypeToken<Map<String, String>>() {
-        }.getType());
+  public ActionSource getSource() {
+    boolean hasRuleIdArg = Optional.ofNullable(args)
+        .filter(map -> map.containsKey(CmdletDescriptor.RULE_ID))
+        .isPresent();
+    return hasRuleIdArg ? ActionSource.RULE : ActionSource.USER;
```
Comment (SidorovGreg) [inline smart-common/src/main/java/org/smartdata/model/ActionInfo.java]: You can continue optional stream: `map(map -> ActionSource.RULE).orElse(ActionSource.USER)`

File: smart-common/src/main/java/org/smartdata/model/CmdletDescriptor.java [hunk 3]
```diff
@@ -115,20 +118,13 @@ public int getActionSize() {
...
-
-    for (int i = 1; i < getActionSize(); i++) {
-      cmdletBuilder.append(" ; ")
-          .append(getActionName(i))
-          .append(" ")
-          .append(formatActionArguments(getActionArgs(i)));
+    for (int i = 0; i < getActionSize(); i++) {
+      String action = actionToString(getActionName(i), getActionArgs(i));
+      cmdletBuilder.add(action);
     }
```
Comment (SidorovGreg) [inline smart-common/src/main/java/org/smartdata/model/CmdletDescriptor.java]: Probably better to use `IntStream`

File: smart-metastore/src/main/java/org/smartdata/metastore/queries/MetastoreQueryExecutor.java [hunk 4]
```diff
@@ -50,11 +50,17 @@ public <T> List<T> execute(MetastoreQuery query, RowMapper<T> rowMapper) {
         query.toSqlQuery(), query.getParameters(), rowMapper);
   }
 
+  public long executeCount(MetastoreQuery query) {
+    Long rowsCount = namedJdbcTemplate.queryForObject(
+        query.toSqlCountQuery(), query.getParameters(), Long.class);
+    return Optional.ofNullable(rowsCount).orElse(0L);
```
Comment (SidorovGreg) [inline smart-metastore/src/main/java/org/smartdata/metastore/queries/MetastoreQueryExecutor.java]: As I remember `queryForObject` throw `EmptyResultDataAccessException` when result is empty

**Review by SidorovGreg — APPROVED** [review]

File: smart-metastore/src/main/java/org/smartdata/metastore/queries/MetastoreQueryExecutor.java (hunk 4 above)
Comment (tigrulya-exe) [reply smart-metastore/src/main/java/org/smartdata/metastore/queries/MetastoreQueryExecutor.java]: Nice catch! fixed

**Review by iamlapa — APPROVED** [review]

**Pull request merged**